### PR TITLE
Prereq check failure tests

### PR DESCRIPTION
**Pull request by eledavi**
Added tests for getType, getBadImports, and getStarterCode

### CODE REVIEW

**Comment by seanlip** [comment]
@eledavi -- please merge master into this branch, and resolve the merge conflicts. Then, check the diff using "Files Changed" on your GitHub PR, make sure it's what you want it to be, then ping us for review.

Thanks!

File: client/domain/QuestionObjectFactorySpec.js [hunk 1]
```diff
@@ -78,6 +85,10 @@ describe('QuestionObjectFactory', function() {
       expect(question.isLastTask(2)).toBe(false);
       expect(question.isLastTask(-1)).toBe(false);
       expect(question.isLastTask(1)).toBe(true);
+      expect(question.isLastTask(question.getTasks.length -
```
Comment (seanlip) [inline client/domain/QuestionObjectFactorySpec.js]: I think I commented on this before, but I still don't understand the point of the "Z - (Z - 1)" construct...

File: client/services/code_evaluators/PythonPrereqCheckServiceSpec.js [hunk 2]
```diff
@@ -317,7 +317,11 @@ describe('PythonPrereqCheckService', function() {
 
       var prereqCheckFailure = PythonPrereqCheckService.checkCode(
         starterCode, userCode);
+
       expect(prereqCheckFailure.isBadImport()).toEqual(true);
+      expect(prereqCheckFailure.getType()).toEqual("badImport");
```
Comment (seanlip) [inline client/services/code_evaluators/PythonPrereqCheckServiceSpec.js]: Let's use single quotes throughout, for consistency.

**Comment by seanlip** [comment]
Thanks @eledavi, just a couple of comments.

File: client/domain/QuestionObjectFactorySpec.js (hunk 1 above)
Comment (eledavi) [inline client/domain/QuestionObjectFactorySpec.js]: I corrected that in a different PR. I'll look into it and see if I can find out why it's doing this.

File: client/services/code_evaluators/PythonPrereqCheckServiceSpec.js (hunk 2 above)
Comment (eledavi) [inline client/services/code_evaluators/PythonPrereqCheckServiceSpec.js]: Done.

**Pull request merged**